lib/reversetunnel/transport.go

```diff
@@ -246,6 +250,11 @@ func (p *transport) start() {
 		// tunnel from the SSH node by dreq.ServerID. We'll need to forward
 		// dreq.Address as well.
 		directAddress = dreq.Address
+
+		if p.trackUserConnection != nil {
+			closed := p.trackUserConnection()
+			defer closed()
```

Comment:
Why the additional variable?

```suggestion
			defer p.trackUserConnection()()
```